engine/src/test/scala/cromwell/engine/EngineFunctionsSpec.scala

```diff
@@ -15,10 +19,57 @@ class EngineFunctionsSpec extends FlatSpec with Matchers {
     val stdFunctions = Seq(
       "stdout", "stderr", "read_lines", "read_tsv", "read_map", "read_object", "read_objects",
       "read_json", "read_int", "read_string", "read_float", "read_boolean", "write_lines",
-      "write_tsv", "write_map", "write_object", "write_objects", "write_json", "size"
+      "write_tsv", "write_map", "write_object", "write_objects", "write_json", "size", "sub"
     )
     stdFunctions.foreach {func =>
       expectFailure(NoFunctions.getFunction(func)(Seq.empty[Try[WdlValue]]))
     }
   }
+
+  "sub" should "replace a string according to a pattern" in {
+    class TestEngineFn extends WdlStandardLibraryImpl {
+      override def fileSystems: List[FileSystem] = List(FileSystems.getDefault)
+      override def glob(path: String, pattern: String): Seq[String] = ???
+    }
+
+    val engineFn = new TestEngineFn
+
+    val table = Table(
+      ("str", "pattern", "replace", "result"),
+      ("somefilename.bam", ".bam$", ".txt", "somefilename.txt"),
+      ("somefilename.bam.bam", ".bam$", ".txt", "somefilename.bam.txt"),
+      ("somefilenamewith.baminside.ext", ".bam$", ".txt", "somefilenamewith.baminside.ext"),
+      ("somefilename", "^some", "other", "otherfilename")
```

Comment:
can you also add a test case for replacing ".bam$" with "" (basically to strip off an extension)?